sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1694,6 +1696,12 @@ class GpuCachedDoublePassWindowIterator(
 
       waitingForFirstPass.foreach(_.close())
       waitingForFirstPass = None
+
+      firstPassIter.foreach(_.foreach(_._1.foreach(_.close())))
```

Comment:
Why are we pulling off of the results from the iterators to close them?  This is not common unless the Iterators are backed by a cache. But then why have it be an Iterator instead of a List or ArrayBuffer or something like that.

Are we risking leaking data if the retry iterator is not fully drained?  If so then there are a number of other bugs that we need to fix.

Reply:
Okay I filed https://github.com/NVIDIA/spark-rapids/issues/8482 for this. Could you add a comment here about it so that we can find a better way to clean this up?